### PR TITLE
Stabilize the `MetricProducer`

### LINKED ISSUE #3599: Stabilize the `MetricProducer`
The opentelemetry-go project is looking to stabilize their SDK implementation. The SDK currently includes an implementation of [the `MetricProducer`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricproducer). This part of the specification, however, is itself not stable. It is marked as experimental.

Given this section of the specification has working implementation in multiple languages (Go, Java, JS, C++, Rust) and has existed for 8 months, can we start the stabilization process?

cc @dashpole

### PR DESCRIPTION
**Pull request by MrAlias**
Resolve #3599

- Stabilize the feature-freeze `MetricProducer`
- Add a comment on when the SDK needs to be prioritized over the `MetricProducer`.

### CODE REVIEW

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -1493,6 +1493,10 @@ libraries to facilitate conversion between delta and cumulative temporalities.
 +-----------------+            +--------------+
 ```
 
+When new OpenTelemetry integrations are added, the SDK is the preferred
```
Comment (jack-berg) [inline specification/metrics/sdk.md:1496]:
```suggestion
When new OpenTelemetry integrations are added, the API is the preferred
```

**Review by jack-berg — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by carlosalberto** [comment]
@MrAlias @reyang Should this be part of 1.25.0? (which is the September release) cc @open-telemetry/specs-metrics-approvers

**Comment by MrAlias** [comment]
> @MrAlias @reyang Should this be part of 1.25.0? (which is the September release) cc @open-telemetry/specs-metrics-approvers

That sounds ideal to me.

**Comment by reyang** [comment]
> > @MrAlias @reyang Should this be part of 1.25.0? (which is the September release) cc @open-telemetry/specs-metrics-approvers
> 
> That sounds ideal to me.

+1

**Review by Bush5580 — APPROVED** [review]

**Pull request merged**